### PR TITLE
Nuxt, baseURL, and SSG fixes

### PR DESCRIPTION
**Pull request by jamesdaniels**
* Handle Nuxt bundled dependencies
* Add fast dev-mode to Nuxt 2
* Handle ssr true/false in a more conventional way
* Call generate when targeting ssr: false
* Import all the integrations, rather than dynamically require
* Leave it to the framework to build an SPA
* Handle Nuxt3 baseURL
* SPAs for Nuxt3 and Angular now respect baseUrl
* If the Nuxt version couldn't be determined, assume Nuxt3
* Nuxt used process.cwd in a couple places, guard against this
* Fix serveOptimizedImages + SPA

### CODE REVIEW

File: src/frameworks/index.ts [hunk 1]
```diff
@@ -107,7 +109,7 @@ const SupportLevelWarnings = {
   ),
 };
 
-export const FIREBASE_FRAMEWORKS_VERSION = "^0.7.0";
+export const FIREBASE_FRAMEWORKS_VERSION = "0.6.1-canary.35a57cb";
```
Comment (jamesdaniels) [inline src/frameworks/index.ts:112]: This is in ^0.8.0 now

Comment (luc122c) [reply src/frameworks/index.ts:112]: Are we able to test this with experiments enabled on the Firebase CLI?

Comment (jamesdaniels) [reply src/frameworks/index.ts:112]: Once merged and cut, yes. If you'd like to help verify the changes in this branch, you can install it manually via `npm i -g github:firebase/firebase-tools#jamesdaniels_nuxtFixes`

File: src/frameworks/nuxt2/index.ts [hunk 2]
```diff
@@ -1,139 +1,116 @@
...
+  // TODO pack this
+  await copy(configFilePath, join(destDir, basename(configFilePath)));
 
   return { packageJson: { ...packageJson }, frameworksEntry: "nuxt" };
 }
+
+export async function getDevModeHandle(cwd: string) {
+  const host = new Promise<string>((resolve) => {
+    const cli = getNodeModuleBin("nuxt", cwd);
+    const serve = spawn(cli, ["dev"], { cwd: cwd });
```
Comment (leoortizz) [inline src/frameworks/nuxt2/index.ts:101]:
```suggestion
    const serve = spawn(cli, ["dev"], { cwd });
```

**Review by leoortizz — APPROVED** [review]
LGTM! Left a couple nits. Tried dev mode and deployments, all working fine.

**Comment by jamesdaniels** [comment]
Going to clean up my use of `!` in a couple spots, manually test the baseUrl w/SPA changes.

**Pull request merged**